tests/unit/conftest.py

```diff
@@ -135,14 +136,40 @@ def restart_savepoint(session, transaction):  # pylint:disable=unused-variable
         ):  # pylint:disable=protected-access
             session.begin_nested()
 
+    set_factory_boy_sqlalchemy_session(session)
```

Comment:
So if you're gonna use `factories.GradingInfo()` your test needs to use the `db_session` fixture or it'll crash. Which can be in an `@pytest.mark.usefixtures("db_session")` or in a module-level `pytestmark = pytest.mark.usefixtures("db_session")`.

Or you can use `factories.GradingInfo.build()` or `build_batch()` if you just need `GradingInfo`'s but don't actually need them in the DB.

In fact LMS always uses the DB session fixture for every test so you don't need to worry about it.

But if we ever fixed it so that not every test uses the DB session, then tests that want to use DB factories would need to depend on it.

Reply:
Would be very nice to stop this from being automatic. We could turn it on for whole directories at first and get more fine grained later.

Reply:
Yep